nixos/modules/programs/rivalcfg.nix

```diff
@@ -0,0 +1,86 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.rivalcfg;
+in {
+  options.programs.rivalcfg = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Small CLI utility program that allows you to configure SteelSeries Rival gaming mice. Add yourself to the "mouse" group to be able to use rivalcfg as a user.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment = {
+      systemPackages = with pkgs; [ rivalcfg ];
+    };
+
+    users.extraGroups.mouse.gid = config.ids.gids.mouse;
+    services.udev.extraRules = ''
+      # SteelSeries Rival
```

Comment:
Where are you sourcing these rules from? It may be better to set up a derivation to fetch these

Reply:
Those rules are from a file in the project itself, modified so that it uses the mouse group instead of just chmodding the device nodes to 0666 because that seemed like a bad idea to me. I wasn't sure how to patch it and add it to the store though (it doesn't get installed by default), so I just hardcoded it.

Reply:
you could use substituteAll or a regular patch to patch it and you could maybe add to postInstall to manually install it?

Reply:
I'll try that.